### ISSUE TITLE
Reading property of undefined

### ISSUE THREAD

**Issue by yekver** [anymaniax/orval#494]
After switching to v6.9.0 I've faced with an error:
```
 TypeError: Cannot read properties of undefined (reading 'properties')
```
for some schemas. Unfortunately there is no ability to run generation in verbose mode to identify where the error is.

**Comment by anymaniax** [comment]
Do you have a schema to reproduce this?

**Comment by anymaniax** [comment]
I will need that to be able to debug. Sorry for the issue

**Comment by yekver** [comment]
> I will need that to be able to debug. Sorry for the issue

Yeah, this is obvious. Are there is a possibility to get some more information about the file:line where  the error has happened?

**Comment by anymaniax** [comment]
Unfortunately not at the moment, I can just tell you that it happened in the generator part.  When orval tries to process the specification. Do you have some discriminators because we added a resolver for that?

**Comment by yekver** [comment]
Yeah, I have one and removing it fixes the issue. Discriminator looks like this:
```yaml
components:
  schemas:
    Shape:
      type: object
      required:
         - name
      properties:
        name:
          $ref: "#/components/schemas/Name"
      discriminator: {
        propertyName: name
        mapping:
          Rectangle: "#/components/schemas/Rectangle"
    
   Rectangle: {
      type: object
      allOf:
        - $ref": "#/components/schemas/Shape"
        - type: object
           properties:
             somePropName:
               type: string

    Name:
      type: string
      readOnly: true
```

**Comment by anymaniax** [comment]
Nice will take a look un try to do a fix asap

**Comment by anymaniax** [comment]
I don't have any issue on my side just that I needed to modify your schema a bit you seem to have a problem with `{` in your schema no?

```yaml
components:
  schemas:
    Shape:
      type: object
      required:
        - name
      properties:
        name:
          $ref: '#/components/schemas/Name'
      discriminator:
        propertyName: name
        mapping:
          Rectangle: '#/components/schemas/Rectangle'

    Rectangle:
      type: object
      allOf:
        - $ref: '#/components/schemas/Shape'
        - type: object
          properties:
            somePropName:
              type: string

    Name:
      type: string
      readOnly: true
```

**Comment by yekver** [comment]
> I don't have any issue on my side just that I needed to modify your schema a bit you seem to have a problem with `{` in your schema no?
> 
> ```yaml
> components:
>   schemas:
>     Shape:
>       type: object
>       required:
>         - name
>       properties:
>         name:
>           $ref: '#/components/schemas/Name'
>       discriminator:
>         propertyName: name
>         mapping:
>           Rectangle: '#/components/schemas/Rectangle'
> 
>     Rectangle:
>       type: object
>       allOf:
>         - $ref: '#/components/schemas/Shape'
>         - type: object
>           properties:
>             somePropName:
>               type: string
> 
>     Name:
>       type: string
>       readOnly: true
> ```


no,  extra `{` is not the case. I'll try to find a way to reproduce this bug

**Comment by maartencardinaels** [comment]
we can confirm that the moment we have a discriminator, the same issue is happening

**Comment by anymaniax** [comment]
@maartencardinaels  do you have a swagger to reproduce the problem?

**Comment by maartencardinaels** [comment]
@anymaniax could you try the example provided [here](https://docs.apimatic.io/changelog/added-support-for-openapi-discriminator-mapping/)?  This should reproduce the error

**Comment by anymaniax** [comment]
ok I have the error thanks

**Comment by missing1984** [comment]
Running into the same issue with discriminator on 6.9.0, downgrade to 6.8.1 works. Thanks for looking into this!

**Comment by anymaniax** [comment]
Fixed will release that asap

**Comment by anymaniax** [comment]
Should be fixed with 6.9.1
